code/modules/power/supermatter/supermatter.dm

```diff
@@ -1050,8 +1050,19 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	//This gotdamn variable is a boomer and keeps giving me problems
 	var/turf/target_turf = get_turf(target)
 	var/pressure = 1
+	// Calculate pressure and do electrolysis.
 	if(target_turf?.return_air())
-		pressure = max(1,target_turf.return_air().return_pressure())
+		var/datum/gas_mixture/air_mixture = target_turf.return_air()
+		pressure = max(1, air_mixture.return_pressure())
+		for(var/reaction in GLOB.electrolyzer_reactions)
+			var/datum/electrolyzer_reaction/current_reaction = GLOB.electrolyzer_reactions[reaction]
+
+			if(!current_reaction.reaction_check(air_mixture = air_mixture, electrolyzer_args = list(ELECTROLYSIS_ARGUMENT_SUPERMATTER_POWER = power_level)))
+				continue
+
+			current_reaction.react(location = target_turf, air_mixture = air_mixture, working_power = zap_str / 200, electrolyzer_args = list(ELECTROLYSIS_ARGUMENT_SUPERMATTER_POWER = power_level)) // Only a portion of the zap_str contributes to working power.
```

Comment:
seems like something we should have a wrapper for

Reply:
Yeah, or actually a gas mixture proc.